clients/drcachesim/simulator/caching_device.h

```diff
@@ -51,18 +51,28 @@
 // We assume we're only invoked from a single thread of control and do
 // not need to synchronize data access.
 
+class snoop_filter_t;
+
 class caching_device_t {
 public:
     caching_device_t();
     virtual bool
     init(int associativity, int block_size, int num_blocks, caching_device_t *parent,
          caching_device_stats_t *stats, prefetcher_t *prefetcher = nullptr,
-         bool inclusive = false, const std::vector<caching_device_t *> &children = {});
+         bool inclusive = false, bool coherent_cache = false, int id_ = -1,
+         snoop_filter_t *snoop_filter_ = nullptr,
+         const std::vector<caching_device_t *> &children = {});
     virtual ~caching_device_t();
     virtual void
     request(const memref_t &memref);
     virtual void
-    invalidate(const addr_t tag);
+    invalidate(const addr_t tag, invalidation_type_t invalidation_type_);
+    bool
+    check_tags(const addr_t &tag_in);
```

Comment:
perhaps call this `contains_tag`?

Also, no need to take the addr_t by reference, it's just an int. Ditto for `child_eviction` and `propagate_write` below.

Reply:
Fixed.